src/ClientGenerator/CodeGenerator.cs

```diff
@@ -65,19 +67,28 @@ public static bool GenerateCode(CodeGenOptions options)
             return !string.IsNullOrWhiteSpace(generatedCode);
         }
 
-        internal static string GenerateSourceForAssembly(Assembly grainAssembly)
+        private static string GenerateSourceForAssembly(Assembly grainAssembly)
         {
             using (var loggerFactory = new LoggerFactory())
             {
                 var config = new ClusterConfiguration();
-                loggerFactory.AddProvider(new ConsoleLoggerProvider(new ConsoleLoggerSettings()));
+                loggerFactory.AddConsole(LogLevel.Trace);
                 var serializationProviderOptions = Options.Create(
                     new SerializationProviderOptions
                     {
                         SerializationProviders = config.Globals.SerializationProviders,
                         FallbackSerializationProvider = config.Globals.FallbackSerializationProvider
                     });
-                var codeGenerator = new RoslynCodeGenerator(new SerializationManager(null, serializationProviderOptions, null, loggerFactory), loggerFactory);
+                var applicationPartManager = new ApplicationPartManager();
+                applicationPartManager.AddFeatureProvider(new BuiltInTypesSerializationFeaturePopulator());
+                applicationPartManager.AddFeatureProvider(new AssemblyAttributeFeatureProvider<GrainInterfaceFeature>());
+                applicationPartManager.AddFeatureProvider(new AssemblyAttributeFeatureProvider<GrainClassFeature>());
+                applicationPartManager.AddFeatureProvider(new AssemblyAttributeFeatureProvider<SerializerFeature>());
+                applicationPartManager.AddApplicationPart(grainAssembly);
+                applicationPartManager.AddApplicationPart(typeof(RuntimeVersion).Assembly);
+                applicationPartManager.AddApplicationPartsFromAppDomain();
```

Comment:
Why add assemblies from the ```AppDomain``` when generating code? During codegen we rely more on the passed in assembly then on what is loaded, not?

Reply:
Adds the code generator assemblies, like Orleans.dll. Instead we could use the AddApplicationPartsFromReferences extension.